### ISSUE TITLE
FileVersionInfo.InternalName on Unix.

### ISSUE THREAD

**Issue by AtsushiKan** [dotnet/corefx#19784]
On Unix, FileVersionInfo.GetVersionInfo() unconditionally sets the FileVersionInfo.InternalName property to be the same as a FileVersionInfo.OriginalFileName property.

On Windows. these are two distinct properties that can be set individually.

**Comment by stephentoub** [comment]
@AtsushiKan, can you elaborate?  What do you expect InternalName to be set to in the Unix implementation? i.e. currently it's set here:
https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs#L84
```C#
_internalName = _originalFilename = Path.GetFileName(_fileName);
```
What are you saying that should be instead?

**Comment by AtsushiKan** [comment]
I don't have a prescription for a fix. This is to track the issue as we have to condition the corefx test onit.

**Comment by stephentoub** [comment]
Let me rephrase.  What is the problem?  In what situation would you expect these to be different on Unix?

**Comment by AtsushiKan** [comment]
It's exposed in the latest revision to PR (https://github.com/dotnet/corefx/pull/19778). Once that's merged, I can put in a more permanent link. The summary version is, you build a managed assembly as "Foo.dll", then use file operations to rename it "Foo.ildll", Windows and Unix report these two properties differently. Windows reports them as "Foo.dll" (which seems more correct to me) and Unix reports them as "Foo.ildll".

**Comment by stephentoub** [comment]
Oh, so it's not that you expect the values to be different from each other, it's that you expect both properties to be the same but to have a different answer on Unix than they're currently giving.

**Comment by stephentoub** [comment]
The Windows implementation is based on support that doesn't exist on Unix, so on Unix we just read metadata from assemblies.  Which piece of metadata do you believe is the right thing to return in this case?  I'm not seeing anything better than what's currently used.

**Comment by AtsushiKan** [comment]
Whatever we return, it probably should be deterministic and based only on the content of the file - not what filename we happened to access it under. Other than that, it's probably not too critical what we return. Maybe returning the AssemblyName (or taking the assembly name and suffixing ".dll") will give most people what they expect?
